source/context.cpp

```diff
@@ -225,7 +225,11 @@ void Context::bind(Config const &config)
 
 		outs() << "Generate bindings, pass " << pass << "...\n";
 
-		for(auto & sp : binders) {
+		// Context::add might invalidate iterators over binders
+		// when called inside b.bind(*this)
+		// => need to loop over a copy here
+		auto binders_ = binders;
```

Comment:
This is problematic from performance perspective: for instance in my test project `binders` will hold ~50k objects so copying it will take notable time. So i would recommend to instead refactor this loop to use plain indexes.